src/mixins/Registry.ts

```diff
@@ -17,18 +15,20 @@ export function RegistryMixin<T extends Constructor<WidgetBase<RegistryMixinProp
 
 		@diffProperty('registry')
 		public diffPropertyRegistry(previousValue: WidgetRegistry, value: WidgetRegistry): PropertyChangeRecord {
+			let changed = false;
+			if (!previousValue) {
+				this.registries.add(value);
+				changed = true;
+			}
+			else if (previousValue !== value) {
+				this.registries.replace(previousValue, value);
+				changed = true;
+			}
```

Comment:
A reply from @matt-gadd mentions that it's not required to compare lengths anymore as if a new item is added then it will invalidate any widget waiting for it. So no need to perform the check here.